lib/dns/dns.go

```diff
@@ -639,3 +640,94 @@ func (s *Server) Stop() {
 func (s *Server) Wait() {
 	s.wg.Wait()
 }
+
+type FQDNs map[string]bool
+
+// ReverseLookup returns a set of FQDNs for ipAddr from nameservers in /etc/resolv.conf
+// /etc/hosts and /etc/nsswitch.conf are ignored by this function
+func ReverseLookup(ipAddr string) (domains FQDNs) {
```

Comment:
is there any way to use trace.Operation here instead direct logging?

Reply:
oh, yeah, that makes sense. I don't see why this wouldn't be possible off the top of my head